src/sm83/decoder.c

```diff
@@ -521,9 +521,17 @@ static int _decodeOperand(struct SM83Operand op, uint16_t pc, char* buffer, int
 	} else {
 		int written;
 		if (op.flags & SM83_OP_FLAG_RELATIVE) {
-			written = snprintf(buffer, blen, "$%04X", pc + (int8_t) op.immediate);
+			if (disassemblyStyle == DISASSEMBLY_STYLE_HEX) {
+				written = snprintf(buffer, blen, "$0x%04X", pc + (int8_t) op.immediate);
+			} else {
+				written = snprintf(buffer, blen, "$%05i", pc + (int8_t) op.immediate);
+			}
 		} else {
-			written = snprintf(buffer, blen, "$%02X", op.immediate);
+			if(disassemblyStyle == DISASSEMBLY_STYLE_HEX) {
+				written = snprintf(buffer, blen, "$0x%02X", pc + (int8_t) op.immediate);
+			} else {
+				written = snprintf(buffer, blen, "$%03i", pc + (int8_t) op.immediate);
```

Comment:
The $ indicates hexadecimal instead of 0x in SM83 assembly. Use no prefix for decimal and only $ for hexadecimal.